classes/manager.php

```diff
@@ -243,4 +245,18 @@ public function is_considered_slow(profile $profile): bool {
         // requirements for all the different timing types (if they exist).
         return true;
     }
+
+    /**
+     * Increments a counter using the approximate counting algorithm.
+     * Can handle up to PHP_INT_SIZE_BITS counts (2^PHP_INT_SIZE_BITS events).
+     *
+     * @param int $current The current count.
+     * @return int The new count.
+     */
+    public static function approximate_increment(int $current): int {
+        // If the nubmer of events is ever expected to be more than 2 billion, a refactor may be needed.
+        $bits = random_int(PHP_INT_MIN, PHP_INT_MAX);
+        return (($bits << (self::PHP_INT_SIZE_BITS - $current)) == 0) ? $current + 1 : $current;
```

Comment:
this is the crux of the whole thing so I think worth spelling it out as an if instead of a ternary and having a comment here with link off to the wikipedia page

Reply:
done